drake/systems/plants/collision/ResultCollector.h

```diff
@@ -0,0 +1,31 @@
+#ifndef DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+#define DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+namespace DrakeCollision {
+class ResultCollector {
+ public:
+  virtual ~ResultCollector() {}
+
+  virtual void addPointPairResult(const PointPair& result);
+
+  inline void addSingleResult(const ElementId idA, const ElementId idB,
+                              const Eigen::Vector3d& ptA,
+                              const Eigen::Vector3d& ptB,
+                              const Eigen::Vector3d& normal,
+                              const double distance) {
+    addPointPairResult(PointPair(idA, idB, ptA, ptB, normal, distance));
+  }
```

Comment:
Sorry I couldn't join this discussion earlier. In this case I believe what matters is how the public API looks after Doxygen gets through with it. If it shows up as `const double distance`, etc. I would forgo the `const` because that is meaningless in an API. If you really want to make these const, you can still do it but you should split the declaration from the definition so that the API only exposes user-meaningful stuff. (That is, the declaration wouldn't have the const's but the definition would.) This is one of those ugly little bugs in the C++ language definition IMO.